packages/amplify-category-api/src/provider-utils/awscloudformation/cdk-stack-builder/apigw-stack-transform.ts

```diff
@@ -186,23 +186,26 @@ export class ApigwStackTransform {
     const overrideFilePath = pathManager.getResourceDirectoryPath(undefined, AmplifyCategories.API, this.resourceName);
     const overrideJSFilePath = path.join(overrideFilePath, 'build', 'override.js');
 
-    const isBuild = await buildOverrideDir(backendDir, overrideFilePath).catch(error => {
-      printer.debug(`Build error : ${error.message}`);
-      return false;
-    });
+    const isBuild = await buildOverrideDir(backendDir, overrideFilePath);
 
     // skip if packageManager or override.ts not found
     if (isBuild) {
-      const { override } = await import(overrideJSFilePath).catch(error => {
+      let override;
+      try {
+        ({ override } = await import(overrideJSFilePath));
+      } catch {
         formatter.list(['No override file found', `To override ${this.resourceName} run "amplify override api"`]);
```

Comment:
shouldn't the value of the formatter be printed to the screen or returned from this function ?

Reply:
Should we use printer here ?

Reply:
Looks like formatter actually prints to the screen. please disregard this comment